test/env/create_k8s_cluster.sh

```diff
@@ -2,29 +2,50 @@
 
 set -o errexit
 
-if [[ $# != 2 ]] ; then
+if [[ $# != 3 ]] ; then
     echo "Usage $(basename "$0") cluster_name insecure_registry_address"
     echo " - cluster_name a name of KIND cluster e.g mycluster"
     echo " - insecure_registry_address address of external insecure registry e.g 192.168.0.2"
+    echo " - number of intended cluster nodes"
     exit 1
 fi
 
 # desired cluster name; default is "kind"
 kind_cluster_name="$1"
 insecure_registry_address="$2"
+number_of_nodes="$3"
 
-# create a cluster with the local registry enabled in containerd
-cat <<EOF | kind create cluster --name "${kind_cluster_name}" --config=-
+kindConfig="$(cat << EOF
 kind: Cluster
 apiVersion: kind.x-k8s.io/v1alpha4
 containerdConfigPatches:
 - |-
   [plugins."io.containerd.grpc.v1.cri".registry.mirrors."registry:5000"]
     endpoint = ["http://registry:5000"]
+networking:
+  disableDefaultCNI: true
+nodes:
 EOF
+)"
+
+node="
+- role: control-plane"
+
+for (( i=0; i<$number_of_nodes; i++ ))
```

Comment:
I would use some yq here to add items to the array
`yq w sample.yaml "b.d[+]" "bar thing"`
However it is just a nit.